File: server/models/searchhistory.js
const mongoose = require("mongoose");
const { faker } = require("@faker-js/faker");
const searchSchema = new mongoose.Schema({
  searchValue: {
    type: String,
  },
  userId: {
    type: String,
    ref: "User",
  },
  date: {
    type: Date,
    default: Date.now,
  },
});
const SearchHistory = mongoose.model("Search", searchSchema);

module.exports = SearchHistory;

File: ecommerce/lib/models/searchHistory.dart
class SearchHistory {
  final String? id;
  final String? searchValue;
  final String? userId;

  SearchHistory({this.id, this.searchValue, this.userId});

  factory SearchHistory.fromJson(Map<String, dynamic> json) {
    return SearchHistory(
        id: json['_id'],
        searchValue: json['searchValue'],
        userId: json['userId']);
  }
}
